apis/Google.Cloud.Firestore/Google.Cloud.Firestore/AggregateQuerySnapshot.cs

```diff
@@ -0,0 +1,61 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using Google.Api.Gax;
+using System;
+
+namespace Google.Cloud.Firestore;
+
+/// <summary>
+/// An immutable snapshot of aggregate query results.
+/// </summary>
+public sealed class AggregateQuerySnapshot : IEquatable<AggregateQuerySnapshot>
+{
+    /// <summary>
+    /// The query producing this snapshot.
+    /// </summary>
+    public AggregateQuery Query { get; }
+
+    /// <summary>
+    /// The time at which the snapshot was read.
+    /// </summary>
+    public Timestamp ReadTime { get; }
+
+    /// <summary>
+    /// Number of documents that matches the query. In-case when explicit limit is provided in the query.

```

Comment:
The second and third sentences should be a single sentence, something like:

"Note that when the query contains a limit, the count is restricted by that limit."

Reply:
I'd also document the conditions in which it will be null (anticipating that actually being feasible...)